src/OptimalControlUtilities/include/BipedalLocomotionControllers/OptimalControlUtilities/PDController.h

```diff
@@ -0,0 +1,130 @@
+/**
+ * @file PDController.h
+ * @authors Giulio Romualdi
+ * @copyright 2019 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_CONTROLLERS_OPTIMAL_CONTROL_UTILITIES_PD_CONTROLLER_H
+#define BIPEDAL_LOCOMOTION_CONTROLLERS_OPTIMAL_CONTROL_UTILITIES_PD_CONTROLLER_H
+
+// iDynTree
+#include <iDynTree/Core/Rotation.h>
+#include <iDynTree/Core/VectorDynSize.h>
+#include <iDynTree/Core/VectorFixSize.h>
+
+namespace BipedalLocomotionControllers
+{
+namespace OptimalControlUtilities
+{
+/**
+ * PDController class is a virtual class that represents a generic PD controller.
+ */
+template <class T, class U, class W> class PDController
```

Comment:
I know I am a pain :grin: , but I would suggest using meaningful names also in template classes.